lib/Sema/CSApply.cpp

```diff
@@ -8217,7 +8217,7 @@ Expr *ExprRewriter::finishApply(ApplyExpr *apply, Type openedType,
         // Resolve into a MakeTemporarilyEscapableExpr.
         auto *args = apply->getArgs();
         assert(args->size() == 2 && "should have two arguments");
-        auto *nonescaping = args->getExpr(0);
+        auto *nonescaping = cs.coerceToRValue(args->getExpr(0));
```

Comment:
I think we should do exactly what `TypeOf` does here, use `coerceCallArguments` before attempting to use them, that would make sure that AST is updated correctly and call gets LoadExpr as well as `nonescaping`.

Reply:
Thank you for the advice.

When I attempted to replicate the behavior of `TypeOf`, another crash occurred in the following code:

```
    // Now create the verification of the withoutActuallyEscaping operand.
    // Either we fail the uniqueness check (which means the closure has escaped)
    // and abort or we continue and destroy the ultimate reference.
    auto isEscaping = SGF.B.createDestroyNotEscapedClosure(
        l, v, DestroyNotEscapedClosureInst::WithoutActuallyEscaping);
    SGF.B.createCondFail(l, isEscaping, "non-escaping closure has escaped");
```
https://github.com/swiftlang/swift/blob/main/lib/SILGen/SILGenExpr.cpp#L6057

I'll investigate the issue. 

Thank you.

Reply:
Thank you for the advice.

When I attempted to replicate the behavior of `TypeOf`, another crash occurred in the following code:

```
// Create verification for the withoutActuallyEscaping operand.
// If the uniqueness check fails (indicating the closure has escaped),
// the process aborts. Otherwise, it continues and destroys the final reference.
auto isEscaping = SGF.B.createDestroyNotEscapedClosure(
    l, v, DestroyNotEscapedClosureInst::WithoutActuallyEscaping);
SGF.B.createCondFail(l, isEscaping, "non-escaping closure has escaped");
```
https://github.com/swiftlang/swift/blob/main/lib/SILGen/SILGenExpr.cpp#L6057

I will investigate the issue. Thank you.

Reply:
I believe I’ve identified the cause of the `SIL verification failed` issue.

It seems that the `withAsync(true)` is always being set:  
https://github.com/swiftlang/swift/blob/main/lib/Sema/TypeOfReference.cpp#L2278  

However, I don’t think this is necessary when the code is not in an async context. To address this, I updated the implementation by passing the `DeclContext` to the `getTypeOfReferenceWithSpecialTypeCheckingSemantics` function. This allowed me to set an `isAsync` flag based on the current context.  

You can check the changes here:  
https://github.com/swiftlang/swift/pull/79238/commits/ebca3f82973648ce187cae0e65e99b2907cd09e4